docs/api-spec.md

```diff
@@ -0,0 +1,322 @@
+| `items`              | `[]Object` | REQUIRED    |
+| `remainingItemCount` | int64      | REQUIRED    |
+| `apiVersion`         | string<br>(`"tekton.dev/v1beta1"`) | REQUIRED    |
+| `kind`               | string<br>(e.g., `"ObjectList"`)   | REQUIRED    |
+
+**NB:** All other fields inherited from [`ListOptions`] or [`ListMeta`]supported by the Kubernetes implementation (e.g., `fieldSelector`,  `labelSelector`) are **OPTIONAL** for the purposes of this spec.
+
+**NB:** The sort order of items returned by list requests is unspecified at this time.
+
+## Detailed Resource Types - v1beta1
+
+### `ArrayOrString`
+
+| Field Name  | Field Type | Requirement |
+|-------------|------------|-------------|
+| `type`      | Enum:<br>- `"string"` (default)<br>- `"array"` | REQUIRED |
+| `stringVal` | string     | REQUIRED    |
+| `arrayVal`  | []string   | REQUIRED    |
+
+### `ContainerStateRunning`
+
+| Field Name   | Field Type | Requirement |
+|--------------|------------|-------------|
+| `startedAt`* | string     | REQUIRED    |
+
+\* `startedAt` MUST be populated by the implementation, in [RFC3339](https://tools.ietf.org/html/rfc3339).
+
+### `ContainerStateWaiting`
+
+| Field Name | Field Type | Requirement |
+|------------|------------|-------------|
+| `reason`   | string     | REQUIRED    |
+| `message`  | string     | REQUIRED    |
+
+### `ContainerStateTerminated`
+
+| Field Name    | Field Type | Requirement |
+|---------------|------------|-------------|
+| `exitCode`    | int32      | REQUIRED    |
+| `reason`      | string     | REQUIRED    |
+| `message`     | string     | REQUIRED    |
+| `startedAt`*  | string     | REQUIRED    |
+| `finishedAt`* | string     | REQUIRED    |
+
+\* `startedAt` and `finishedAt` MUST be populated by the implementation, in [RFC3339](https://tools.ietf.org/html/rfc3339).
+
+### `EnvVar`
+
+| Field Name | Field Type | Requirement |
+|------------|------------|-------------|
+| `name`     | string     | REQUIRED    |
+| `value`    | string     | REQUIRED    |
+
+**NB:** All other [EnvVar](https://godoc.org/k8s.io/api/core/v1#EnvVar) types inherited from [core.v1/EnvVar](https://godoc.org/k8s.io/api/core/v1#EnvVar) and supported by the Kubernetes implementation (e.g., `valueFrom`) are **OPTIONAL** for the purposes of this spec.
+
+### `Param`
+
+| Field Name | Field Type      | Requirement |
+|------------|-----------------|-------------|
+| `name`     | string          | REQUIRED    |
+| `value`    | `ArrayOrString` | REQUIRED    |
+
+### `ParamSpec`
+
+| Field Name    | Field Type | Requirement |
+|---------------|------------|-------------|
+| `name`        | string     | REQUIRED    |
+| `description` | string     | REQUIRED    |
+| `type`        | Enum: <br>- `"string"` (default) <br>- `"array"` | REQUIRED |
+| `default`     | `ArrayOrString` | REQUIRED |
+
+### `Step`
+
+| Field Name   | Field Type | Requirement |
+|--------------|------------|-------------|
+| `name`       | string     | REQUIRED    |
```

Comment:
am i crazy or is it possible to have a step without a name

Reply:
This doc is saying it's "required" that platform implementations _support_ naming steps, not that the user's request _must_ specify one.

If there's a good clear way to describe that somewhere I'd love to clear that up for future readers.